gcp_variant_transforms/beam_io/vcf_parser.py

```diff
@@ -504,6 +570,224 @@ def _get_variant_calls(self, record, formats):
+
+    # Pysam pipe is responsible for supplying lines from child process into the
+    # VariantFile, since it can only take an actual file descriptor as an input.
+    pysam_read, pysam_write = os.pipe()
+    # Since child process doesn't have access to the lines that need to be
+    # parsed, variants pipe is needed to supply them from _get_variant() method
+    # into the child process, to be propagated afterwards into the pysam pipe.
+    variants_read, variants_write = os.pipe()
+    pid = os.fork()
+    if pid:
+      self._process_pid = pid
+      self._init_parent_process(pysam_read, variants_write)
+    else:
+      self._init_child_process(variants_read, pysam_write, header_lines)
+
+  def _get_variant(self, data_line):
+    try:
+      self._to_child.write(data_line.encode('utf-8') + '\n')
+      self._to_child.flush()
+      return self._convert_to_variant(next(self._vcf_reader))
+    except (MemoryError, IOError, ValueError) as e:
+      logging.warning('VCF record read failed in %s for line %s: %s',
+                      self._file_name, data_line, str(e))
+      return MalformedVcfRecord(self._file_name, data_line, str(e))
+
+  def _verify_record(self, record):
+    if record.start < 0:
+      raise ValueError('Start position is incorrect.')
+    if record.stop < 0:
+      raise ValueError('End position is incorrect.')
+
+  def _convert_to_variant(self, record):
+    # type: (libcbcf.VariantRecord) -> Variant
+    """Converts the PySAM record to a :class:`Variant` object.
+
+    Args:
+      record: An object containing info about a variant.
+
+    Returns:
+      A :class:`Variant` object from the given record.
+
+    Raises:
+      ValueError: if `record` is semantically invalid.
+    """
+    self._verify_record(record)
+    return Variant(
+        reference_name=record.chrom.encode('utf-8'),
+        start=record.start,
+        end=record.stop,
+        reference_bases=self._convert_field(record.ref),
+        alternate_bases=list(record.alts) if record.alts else [],
+        names=record.id.split(';') if record.id else [],
+        quality=self._parse_to_numeric(record.qual) if record.qual else None,
+        filters=(None if not list(record.filter.keys()) else
+                 list(record.filter.keys())),
+        info=self._get_variant_info(record),
+        calls=self._get_variant_calls(record.samples))
+
+  def _get_variant_info(self, record):
+    # type: (libcbcf.VariantRecord) -> Dict[str, Any]
+    info = {}
+    for k, v in list(record.info.items()):
+      if k != END_INFO_KEY:
+        if isinstance(v, tuple):
+          info[k] = list(map(self._convert_field, v))
+        else:
+          # If a field was not provided in the header, make it a list to
+          # follow the PyVCF standards.
+          info[k] = (
+              self._convert_field(v) if k in self._original_info_list else
+              [self._convert_field(v)])
+
+    return info
+
+  def _parse_to_numeric(self, value):
```

Comment:
What is this function meant to do?  A comment on the function would be very helpful

It looks like it is trying a variety of conversions to coerce a string (or number?) value into another type of numeric value

Is there no python built in functions for this?